azurerm/internal/services/digitaltwins/digital_twins_instance_resource.go

```diff
@@ -59,23 +59,25 @@ func resourceArmDigitalTwins() *schema.Resource {
 		},
 	}
 }
-func resourceArmDigitalTwinsCreate(d *schema.ResourceData, meta interface{}) error {
+func resourceArmDigitalTwinsInstanceCreate(d *schema.ResourceData, meta interface{}) error {
 	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
-	client := meta.(*clients.Client).IoTHub.DigitalTwinsClient
+	client := meta.(*clients.Client).DigitalTwins.InstanceClient
 	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
 	name := d.Get("name").(string)
 	resourceGroup := d.Get("resource_group_name").(string)
 
+	id := parse.NewDigitalTwinsInstanceID(subscriptionId, resourceGroup, name)
+
 	existing, err := client.Get(ctx, resourceGroup, name)
 	if err != nil {
 		if !utils.ResponseWasNotFound(existing.Response) {
-			return fmt.Errorf("checking for present of existing Digital Twins %q (Resource Group %q): %+v", name, resourceGroup, err)
+			return fmt.Errorf("checking for existing Digital Twins Instance %q (Resource Group %q): %+v", name, resourceGroup, err)
 		}
 	}
 	if existing.ID != nil && *existing.ID != "" {
-		return tf.ImportAsExistsError("azurerm_iothub_digital_twins", *existing.ID)
+		return tf.ImportAsExistsError("azurerm_digital_twins_instance", *existing.ID)
 	}
```

Comment:
we can then use that here:

```suggestion
	if !utils.ResponseWasNotFound(existing.Response) {
		return tf.ImportAsExistsError("azurerm_digital_twins_instance", id)
	}
```